monai/apps/auto3dseg/hpo_gen.py

```diff
@@ -0,0 +1,263 @@
+        """Return the dumped pickle object of algo"""
+        return self.obj_filename
+
+    def print_nni_instruction(self):
+        """
+        Print how to write the trial commands in NNI
+        """
+        hint = "python -m monai.apps.auto3dseg NNIGen run_algo "
+        logger.info("=" * 140)
+        logger.info("If NNI will run in your local env: ")
+        logger.info("1. Add the following line to the trialCommand in your NNI config: ")
+        logger.info(f"{hint} {self.obj_filename} {{result_dir}}")
+        logger.info("-" * 140)
+        logger.info("If NNI will run in a remote env: ")
+        logger.info(
+            f"1. Copy the algorithm_templates folder {self.algo_templates_dir} to remote {{remote_algorithm_templates_dir}}"
+        )
+        logger.info(f"2. Copy the base_task folder {self.base_task_dir} to the remote machine {{remote_base_task_dir}}")
+        logger.info("Then add the following line to the trialCommand in your NNI config: ")
+        logger.info(f"{hint} {{remote_base_task_dir}} {{result_dir}} {{remote_algorithm_templates_dir}}")
+        logger.info("=" * 140)
+
+    def get_hyperparameters(self):
+        """
+        Get parameter for next round of training from nni server
+        """
+        if has_nni:
+            return nni.get_next_parameter()
+        else:
+            return {}
+
+    def update_params(self, params: dict):  # type: ignore
+        """
+        Translate the parameter from monai bundle to nni format.
+
+        Args:
+            params: a dict of parameters.
+        """
+        self.params = params
+
+    def get_task_id(self):
+        """
+        Get the identifier of the current experiment. In the format of listing the searching parameter name and values
+        connected by underscore in the file name.
+        """
+        task_id = ""
+        for k, v in self.params.items():
+            task_id += f"_{k}_{v}"
+        if len(task_id) == 0:
+            task_id = "_None"  # avoid rewriting the original
+        return task_id
+
+    def generate(self, output_folder: str = ".") -> None:
+        """
+        Generate the record for each Algo. If it is a BundleAlgo, it will generate the config files.
+
+        Args:
+            output_folder
+        """
+        task_id = self.get_task_id()
+        task_prefix = os.path.basename(self.algo.get_output_path())
+        write_path = os.path.join(output_folder, task_prefix + task_id)
+        self.obj_filename = os.path.join(write_path, "algo_object.pkl")
+
+        if isinstance(self.algo, BundleAlgo):
+            self.algo.export_to_disk(output_folder, task_prefix + task_id)
+            algo_export_to_pickle(self.algo, self.obj_filename, self.algo_templates_dir)
+        else:
+
+            ConfigParser.export_config_file(self.params, write_path)
+            logger.info(write_path)
+
+    def set_score(self, acc):
+        """
+        Report the acc to nni server
+        """
+        if has_nni:
```

Comment:
I feel no need to check `if has_nni`, because this `NNIGen` must work with NNI, right?

Thanks.

Reply:
Our test system doesn't have nni, but the `run_algo` worth testing and can be tested. Also, the user can try it locally without nni. So that's why I check nni here.

Reply:
OK, please give a warning if no NNI.

Thanks.

Reply:
Done. Thanks!